src/Nox.Generator/Common/TemplateBuilderBase.cs

```diff
@@ -71,9 +70,9 @@ public TemplateBuilderBase WithFileNameSuffix(string fileNameSuffix)
     /// <summary>
     /// Extend the default model with a extended property to the extendedModel
     /// </summary>
-    public TemplateBuilderBase WithObject(string name, object value)
+    public TemplateBuilderBase WithObject(string name, object? value)
     {
-        _model[name] = value;
+        if (value != null) _model[name] = value;
```

Comment:
IMO we should not accept nulls, the calling method should not set the object if null in this case
```suggestion
ArgumentNullException.ThrowIfNull(value);
   _model[name] = value;
```

Reply:
Implemented